### PR TITLE
iio-sensor-proxy: 3.0 -> 3.5, cleanup

### PR DESCRIPTION
**Pull request by 999eagle**
###### Description of changes

Changelog between 3.0 and 3.5 can be found here: https://gitlab.freedesktop.org/hadess/iio-sensor-proxy/-/blob/master/NEWS

###### Things done

I've also tested this update on my librem5 to ensure that display rotation still works.

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by peterhoeg** [comment]
Do you mind dropping me from the maintainer list? I no longer have the required hardware.

**Comment by 999eagle** [comment]
Okay, I've set myself as the maintainer.

File: pkgs/os-specific/linux/iio-sensor-proxy/default.nix [hunk 1]
```diff
@@ -1,51 +1,57 @@
...
-    runHook preConfigure
-
-    ./autogen.sh --prefix=$out \
-      --with-udevrulesdir=$out/lib/udev/rules.d \
-      --with-systemdsystemunitdir=$out/lib/systemd/system
-
-    runHook postConfigure
+  postPatch = ''
+    # upstream meson.build currently doesn't have an option to change the default polkit dir
+    sed -i -Ee "s|polkit_policy_directory|'$out/share/polkit-1/actions'|g;" data/meson.build
```
Comment (Janik-Haag) [inline pkgs/os-specific/linux/iio-sensor-proxy/default.nix:28]: you shuold use `substituteInPlace` here instead of `sed`

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 245773` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>CuboCore.coretoppings</li>
    <li>iio-sensor-proxy</li>
  </ul>
</details>

**Review by Janik-Haag — COMMENTED** [review]
LGTM; except for the nitpick and I don't have the hardware to test this but would be fine with merging if you say it works fine since you will be the new package maintainer.

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Pull request merged**